docs/design/2022-11-17-ttl-table.md

```diff
@@ -0,0 +1,357 @@
+# Proposal: Support TTL Table
+- Author(s): [lcwangchao](https://github.com/lcwangchao)
+- Tracking Issue: https://github.com/pingcap/tidb/issues/39262
+
+## Table of Contents
+
+<!-- TOC -->
+* [Proposal: Support TTL Table](#proposal--support-ttl-table)
+  * [Table of Contents](#table-of-contents)
+  * [Introduction](#introduction)
+  * [Detailed Design](#detailed-design)
+    * [Syntax](#syntax)
+      * [Create TTL Table](#create-ttl-table)
+      * [Alter a Table with TTL](#alter-a-table-with-ttl)
+      * [Alter to a non-TTL Table](#alter-to-a-non-ttl-table)
+      * [Constraints](#constraints)
+    * [TTL Job Management](#ttl-job-management)
+    * [TTL Job Details](#ttl-job-details)
+      * [Scan Tasks](#scan-tasks)
+      * [Delete Tasks](#delete-tasks)
+    * [New System Variables](#new-system-variables)
+    * [New Metrics](#new-metrics)
+  * [Known Issues](#known-issues)
+  * [Future Works](#future-works)
+  * [Alternative Solutions](#alternative-solutions)
+<!-- TOC -->
+
+## Introduction
+
+The rows in a TTL table will be deleted automatically when they are expired. It is useful for some scenes, for example, delete the expired verification codes. A TTL table will have a column with the type DATE/DATETIME/TIMESTAMP which will be compared with the current time, if the interval between them exceeds some threshold, the corresponding row will be deleted.
+
+## Detailed Design
+
+### Syntax
+
+#### Create TTL Table
+
+The following example shows how to create a TTL table. The column `create_at` is used to specify the creation time of the rows which will be deleted 3 months after that.
+
+```sql
+CREATE TABLE t1 (
+    id int PRIMARY KEY,
+    created_at TIMESTAMP
+) TTL = `created_at` + INTERVAL 3 MONTH;
+```
+
+We can use another `TTL_ENABLE` option to disable/enable the TTL job for the table. For example:
+
+```sql
+CREATE TABLE t1 (
+    id int PRIMARY KEY,
+    created_at TIMESTAMP
+) TTL = `created_at` + INTERVAL 3 MONTH TTL_ENABLE = 'OFF';
+```
+
+The above table will not delete expired rows automatically because `TTL_ENABLE` is set to `OFF`. When the `TTL_ENABLE` is omitted, it uses value `ON` by default.
+
+To make it compatible with mysql, TTL options also support the comment format. For example:
+
+```sql
+CREATE TABLE t1 (
+    id int PRIMARY KEY,
+    created_at TIMESTAMP
+) /*T![ttl] TTL = `created_at` + INTERVAL 3 MONTH TTL_ENABLE = 'OFF'*/;
+```
+
+#### Alter a Table with TTL
+
+We can alter an exist table with TTL options, for example:
+
+```sql
+ALTER TABLE t1 TTL = `created_at` + INTERVAL 3 MONTH;
+```
+
+We should allow to update the existed TTL options. When it is updated, the running background job for this table should stop or restart according to the newest settings.
+
+#### Alter to a non-TTL Table
+
+If we want to remove a table's TTL options, we can just do:
+
+```sql
+ALTER TABLE t1 REMOVE TTL;
+```
+
```

Comment:
Maybe just add a comment: `REMOVE TTL` is consistent with `REMOVE PARTITIONING`.

(On first read I was going to suggest `NO TTL`, but I think you are more correct here. A comment will let others know it's consistent with existing usage.)